src/main/kotlin/skytils/skytilsmod/listeners/ChatListener.kt

```diff
@@ -279,23 +281,21 @@ class ChatListener {
                 SkillUtils.calcXpWithProgress(catacombsObj.experience ?: 0.0, SkillUtils.dungeoneeringXp.values)
 
             val secrets = playerResponse.achievements.getOrDefault("skyblock_treasure_hunter", 0)
-
+            val name = playerResponse.rankPrefix + if(playerResponse.rank.toString() == "NONE") {""} else {" "} + playerResponse.player.getString("displayname")
             val component = UMessage("&2&l-----------------------------\n")
                 .append(
-                    UTextComponent(playerResponse.rankPrefix
-                            + if(playerResponse.rank.toString() == "NONE") {""} else {" "}
-                            + "${playerResponse.player.getString("displayname")}'s §dStats: §7(Hover)\n").setHoverText("§7Click to run: /skytilscata $username").setClick(
+                    UTextComponent("$name's §dStats: §7(Hover)\n").setHoverText("§7Click to run: /skytilscata $username").setClick(
                         ClickEvent.Action.RUN_COMMAND, "/skytilscata $username")
                 )
                 .append("§dCata: §b${NumberUtil.nf.format(floor(cataLevel))}\n\n")
             val armor = profileData.armor
             val armorArray:ArrayList<List<String>> = ArrayList()
             armor?.forEveryItem {
-                val armorLore = it.asMinecraft.getTooltip(mc.thePlayer,false)
+                val armorLore = it.asMinecraft.getTooltip(mc.thePlayer, false)
                 armorArray.add(armorLore)
             }
             armorArray.reverse()
-            for(armorPiece in armorArray) {
+            for (armorPiece in armorArray) {
```

Comment:
Use `joinToString` instead of a for loop

Reply:
thanks, didn't know that existed in kotlin

Reply:
Half fixed